### PR TITLE
ENCD-3086-add-dataset-to-file

### PR DESCRIPTION
**Pull request by keenangraham**
Add some dataset properties to file via calculated_properties. Will have to see how long this takes to index.

### CODE REVIEW

File: src/encoded/types/file.py [hunk 1]
```diff
@@ -476,6 +479,33 @@ def s3_uri(self):
...
+        return dataset.get('biosample_ontology')
+
+    @calculated_property(schema={
+        "title": "Target",
+        "type": "string",
+        "notSubmittable": True,
+    })
+    def target(self, request, dataset):
+        dataset = request.embed(dataset, '@@object')
+        return dataset.get('target')
```
Comment (hitz) [inline src/encoded/types/file.py]: I think this should just be target.label?

Comment (keenangraham) [reply src/encoded/types/file.py]: K, thought maybe there was some other interesting stuff in target but should probably try to keep it as slim as possible.

File: src/encoded/types/utils.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (hitz) [inline src/encoded/types/utils.py]: are there other functions in types/* that could be moved here?

File: src/encoded/tests/test_types_utils.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
+import pytest
+
+
+@pytest.fixture
+def fastq_no_replicate(award, experiment, lab, platform1):
```
Comment (hitz) [inline src/encoded/tests/test_types_utils.py]: this is duplicated from test_file.py right?   Probably their should be a single fastq fixture in a shared place;  datafixtures.py?  Or if it's distinct, it should be a different name.

Comment (keenangraham) [reply src/encoded/tests/test_types_utils.py]: I pulled duplicates into datafixtures. Works fine.

File: src/encoded/types/utils.py (hunk 2 above)
Comment (keenangraham) [reply src/encoded/types/utils.py]: Hmm. I think I might just put these in https://github.com/ENCODE-DCC/snovault/blob/dev/src/snovault/util.py

**Comment by keenangraham** [comment]
Moved utils to snovault branch.

**Pull request merged**